cli/src/lib/compiler/entrypoints.js

```diff
@@ -0,0 +1,72 @@
+const path = require('path')
+const { reporter, chalk } = require('@dhis2/cli-helpers-engine')
+const fs = require('fs-extra')
+const { normalizeExtension } = require('./extensionHelpers.js')
+
+const verifyEntrypoint = ({ entrypoint, basePath, resolveModule }) => {
+    if (!entrypoint.match(/^(\.\/)?src\//)) {
+        const msg = `Entrypoint ${chalk.bold(
+            entrypoint
+        )} must be located within the ${chalk.bold('./src')} directory`
+        reporter.error(msg)
+        throw new Error(msg)
+    }
+
+    try {
+        resolveModule(path.join(basePath, entrypoint))
+    } catch (e) {
+        const msg = `Could not resolve entrypoint ${chalk.bold(entrypoint)}`
+        reporter.error(msg)
+        throw new Error(msg)
+    }
+}
+
+exports.verifyEntrypoints = ({
+    config,
+    paths,
+    resolveModule = require.resolve,
```

Comment:
This was required as it is not currently possible to mock `require.resolve` using jest